### PR TITLE
Added COVID19 Chatbot Token service to vets API

### PR DESCRIPTION
**Pull request by shaunhubbard-va**
## Description of change
This change set will add a service endpoint to issue tokens for the COVID 19 chatbot.

## Original issue(s)
department-of-veterans-affairs/covid19-chatbot/issues/91

## Things to know about this PR
Three new values were added to `settings.yml`, the two secrets vary by environment and their key will be entered into credstash.

An `rspec` file was added in order to test the new endpoint `rpec/controllers/v0/coronavirus_chatbot_controller_spec.rb`, and two cassettes were made in order to facilitate responses from the external service.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -288,6 +288,8 @@
         defaults: { feature: feature }
       )
     end
+
+    resources :coronavirus_chatbot, only: :create
```
Comment (karlbrown-va) [inline config/routes.rb:292]: should the route be more like coronoavirus_chatbot_tokengen or something? given that this endpoint is mainly just returning the valid token as per my understanding.

Comment (shaunhubbard-va) [reply config/routes.rb:292]: how about `/v0/coronoavirus_chatbot/tokens` so that we can keep it railsy?

Comment (karlbrown-va) [reply config/routes.rb:292]: sure - I take it tokens is better than token re: rails conventions?

Comment (shaunhubbard-va) [reply config/routes.rb:292]: typically resources are pluralized as to reflect the real world, depening on inflector conventions. TL;DR yes

File: config/settings.yml [hunk 2]
```diff
@@ -506,3 +506,8 @@ vet_verification:
 github_stats:
   username: github-stats-rake
   token: fake_token
+
+coronavirus_chatbot:
+  directline_uri: northamerica.directline.botframework.com
```
Comment (johnpaulashenfelter) [inline config/settings.yml:511]: Alphabetical if you don't mind please :)

File: config/settings.yml [hunk 3]
```diff
@@ -506,3 +506,9 @@ vet_verification:
 github_stats:
   username: github-stats-rake
   token: fake_token
+
+# Settings for the coronavirus chatbot tokens
+coronavirus_chatbot:
+  app_secret: a_secret
```
Comment (karlbrown-va) [inline config/settings.yml:512]: consider calling this fake-secret

File: config/settings.yml [hunk 4]
```diff
@@ -506,3 +506,9 @@ vet_verification:
...
+  directline_uri: northamerica.directline.botframework.com
+  webchat_secret: another_secret
```
Comment (karlbrown-va) [inline config/settings.yml:514]: same here, to match the pattern elsewhere

File: app/controllers/v0/coronavirus_chatbot/tokens_controller.rb [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+module V0
+  module CoronavirusChatbot
+    class TokensController < ApplicationController
+      skip_before_action :authenticate, only: [:create]
+
+      rescue_from 'V0::CoronavirusChatbot::TokensController::ServiceException', with: :service_exception_handler
+      rescue_from Net::HTTPError, with: :service_exception_handler
+
+      def create
```
Comment (karlbrown-va) [inline app/controllers/v0/coronavirus_chatbot/tokens_controller.rb:11]: what sort of authentication is on this endpoint? Can anyone hit it or only va.gov front end?

Comment (shaunhubbard-va) [reply app/controllers/v0/coronavirus_chatbot/tokens_controller.rb:11]: per line 6  in this same file, none

**Pull request merged**